controller.py

```diff
@@ -310,6 +312,63 @@ def controller(
     if istart != -1:
         faultlist = faultlist[istart:]
 
+    # Remove already simulated faults
+    if missing_only:
+        clogger.info("Identifying already simulated faults")
+
+        hdf5_file = Path(hdf5path)
+        if hdf5_file.is_file():
```

Comment:
I would prefer to have the error handling first and then an `else` with the logic.